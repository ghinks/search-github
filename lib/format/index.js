'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _table = require('table');

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const output = nodes => {
  let data = nodes.map(node => [node.number, _moment2.default.duration(_moment2.default.utc(node.publishedAt).diff((0, _moment2.default)())).humanize(), node.title]);
  data = [['Issue #', 'when', 'Title'], ...data];
  const config = {
    columns: {
      2: {
        width: 70,
        truncate: 100
      }
    }
  };
  return (0, _table.table)(data, config);
};

exports.default = output;