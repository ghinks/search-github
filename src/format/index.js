import { table } from 'table'
import moment from 'moment'

const output = (nodes) => {
  let data = nodes.map(node => [node.number, moment.duration(moment.utc(node.publishedAt).diff(moment())).humanize(), node.title])
  data = [['Issue #', 'when', 'Title'], ...data]
  return table(data)
}

export default output
