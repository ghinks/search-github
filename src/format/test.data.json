{
  "nodes": [
    {
      "author": {
        "login": "keyiis"
      },
      "publishedAt": "2018-05-08T03:17:07Z",
      "id": "MDU6SXNzdWUzMjEwMjgzNjg=",
      "title": "Can't throw error in router middleware ",
      "number": 1234,
      "body": "abc"
    }
  ]
}