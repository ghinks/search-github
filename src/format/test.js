import format from './index'
import fsp from 'fs/promises'
import moment from 'moment'

describe('format', () => {
  let nodes

  const setPublishedAt = () => new Date(moment().subtract(7, 'days'))

  beforeAll((done) => {
    fsp.readFile(`${__dirname}/test.data.json`, 'utf8')
      .then((testData) => {
        nodes = JSON.parse(testData)['nodes']
        nodes[0].publishedAt = setPublishedAt()
        done()
      })
  })
  test('expect formated output', () => {
    expect(format(nodes)).toMatchSnapshot()
  })
})
