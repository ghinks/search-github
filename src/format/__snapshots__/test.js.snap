// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format expect formated output 1`] = `
"╔═════════╤═════════╤═════════════════════════════════════════╗
║ Issue # │ when    │ Title                                   ║
╟─────────┼─────────┼─────────────────────────────────────────╢
║ 1234    │ 13 days │ Can't throw error in router middleware  ║
╚═════════╧═════════╧═════════════════════════════════════════╝
"
`;
